src/app/modules/game/pages/lobby/lobby.component.ts

```diff
@@ -1,43 +1,68 @@
-import {Component, OnInit} from '@angular/core';
-import {SocketService} from '../../services/socket.service';
-import {ISocket} from '../../interfaces/isocket';
-import {DataStoreService} from '../../../../core/services/data-store.service';
+import { Component, OnInit } from "@angular/core";
+import { SocketService } from "../../services/socket.service";
+import { ISocket } from "../../interfaces/isocket";
+import { DataStoreService } from "../../../../core/services/data-store.service";
+import { Router } from "@angular/router";
 
 @Component({
-  selector: 'app-lobby',
-  templateUrl: './lobby.component.html',
-  styleUrls: ['./lobby.component.scss'],
+  selector: "app-lobby",
+  templateUrl: "./lobby.component.html",
+  styleUrls: ["./lobby.component.scss"]
 })
 export class LobbyComponent implements OnInit {
   public users: string[] = [];
   public roomCode: string;
   public errorMessage: string;
   public username: string;
+  public gameStarted: boolean;
 
-  constructor(private socketService: SocketService, private dataStore: DataStoreService) {
+  constructor(
+    private socketService: SocketService,
+    private dataStore: DataStoreService,
+    private router: Router
+  ) {
     this.roomCode = this.dataStore.getRoomCode();
     this.username = this.dataStore.getUserName();
+    this.gameStarted = false;
   }
 
   ngOnInit(): void {
     this.configSocketListeners();
   }
 
   private configSocketListeners(): void {
-    this.socketService.listen('new-user-connected').subscribe((data: ISocket) => {
-      this.users = [...data.payload];
-    });
-    this.socketService.emit('new-user', {username: this.username, room: this.roomCode});
-    this.socketService.listen('user-disconnected').subscribe((data: ISocket) => {
-      this.users = this.users.filter(
-        (username: string) => username !== data.payload.username
-      );
+    this.socketService
+      .listen("new-user-connected")
+      .subscribe((data: ISocket) => {
+        this.users = [...data.payload];
+
+        if (this.users.length >= 3 && this.users.length <= 6) {
```

Comment:
you can disable button if users array length < 3 in template, not on subscribe

Reply:
Agreed with @maxonchil to leave as is. We couldn`t identify the number of users in html, because the logic is used in many cases.